Revert "E2E: Temporary remove block editor test from IE11 canary suite"

ockham opened the pull request:
Reverts Automattic/wp-calypso#40510.

This should be obsolete per D40956-code.

**Testing Instructions**

Verify that the IE11 Canary goes green on this PR.

Resolves #40502

ockham committed:
Revert "E2E: Temporary remove block editor test from IE11 canary suite (#40510)"

This reverts commit f4dcb62e27b3e8fbc83d531c0402a3ac31f854a0.

mmtr commented:
Seems the E2E tests cannot find the iframe witch matches with what I see when testing on Browserstack: Calypso cannot load Gutenframe so it redirects to WP Admin.

![Mar-27-2020 16-56-58](https://user-images.githubusercontent.com/1233880/77774862-057b1800-704c-11ea-810f-366448313b62.gif)

ockham commented:
> Seems the E2E tests cannot find the iframe which matches with what I see when testing on Browserstack: Calypso cannot load Gutenframe so it redirects to WP Admin.
> 
> ![Mar-27-2020 16-56-58](https://user-images.githubusercontent.com/1233880/77774862-057b1800-704c-11ea-810f-366448313b62.gif)

Can also repro :confused:

gwwar commented:
@ockham did you have time for this one, or should we hand off to @roo2 and @yansern ?

simison commented:
If @roo2 and @yansern you can take a look tomorrow, that'd be great! cc @ramonjd 

Thanks for the nudge, @gwwar :-)

gwwar commented:
Can we please make sure this test is re-enabled before we need to test the next round of updates?

ramonjd commented:
> Can we please make sure this test is re-enabled before we need to test the next round of updates?

@roo2 Have you got some time to look at this issue today? I think @yansern has been caught in a travel situation.

roo2 commented:
👀  Ahh yep true I'll get onto this today.

ockham commented:
Thanks for picking this up @roo2!

p-jackson commented:
Decided that getting IE to use the iframe editor is a won't fix
p1585885414030400-slack-C7YPUHBB2?thread_ts=1585885414.030400&cid=C7YPUHBB2

Plan B is going to be fixing the tests so they work in the wp-admin editor.

yansern commented:
Yes the decision is to fix/remove the E2E test instead of the redirection bug.

While working on this issue I stumbled upon many roadblocks related to IE dev workflow, more information in p4TIVU-9nX-p2.

gwwar commented:
@yansern @ramonjd @ockham  I would really recommend not removing this test until core adds in their own suites for IE. Basic things break all the time.

noahtallen commented:
Cool. I will try to update this test to work with the redirection, then. (if I'm understanding everything correctly.)

simison commented:
Yep, so in short;
- ok if IE11 users end up in wp-admin
- E2E tests need to accommodate for that

p-jackson committed:
GutenbergEditorComponent falls back to wp-admin editor

If the iframe on the editor page can't be found then it will assume it
is on the wp-admin editor page. An issue with IE means users are being
redirected to the wp-admin editor, and this allows e2e tests to pass in
that situation as users are still able to achieve what they want to
achieve.

p-jackson commented:
I've updated `GutenbergEditorComponent` so that by default it works regardless of whether it's using the wp-admin or iframe'd editor.

This fixes the IE canary tests because for some reason the block editor has started loading in wp-admin for IE users.

I think having the `GutenbergEditorComponent` work for either editor makes sense since page objects are supposed to abstract over technical details. From a users point of view the tests are able to achieve the things they're written to achieve.

However it does mean we're no longer testing that evergreen browsers load the iframe editor. Is this something we should worry about testing in an e2e tests? If so should we just have a single non-iecanary test that explicitly checks for the iframe editor i.e. `GutenbergEditorComponent.Expect( driver, 'iframe' )`?

Ping @bsessions85 @Stojdza since changing all the tests is a big deal.

p-jackson commented:
`ci/wp-e2e-tests-full-desktop-edge` is failing, but I've removed the "needs gutenberg edge" label. We'll address edge failures here #40451

noahtallen reviewed (approved):
This seems reasonable to me. Essentially, it allows any test to fall back to wp-admin if it cannot find the iFrame (e.g. just like IE11 is doing). This is an "ok" user experience, so I think the tests should pass.

theoretically, we could also just do:

```
if ( isIe11 ) return;
```

instead of passing the argument, or we could set the `editorType` through the test config, but I have no idea how to do that.

p-jackson commented:
I'll merge this we can catch regressions in new Gutenberg regressions. But still interested in any feedback from @Stojdza / @bsessions85 if this we should limit to IE only or something like that.

simison commented:
>  However it does mean we're no longer testing that evergreen browsers load the iframe editor. Is this something we should worry about testing in an e2e tests?

Yep, if one of the new browser's starts redirecting to WP admin unplanned, it would be an issue. Let's add the test for that case.

Nice job here!

p-jackson commented:
Adding a test that asserts the editor opens in the iframe here #40832

ockham commented:
Still failing across new PRs(e.g. #40807, #40841, #40842), _and_ on the follow-up PR that's supposed to address this (#40832). Going to revert the revert :confused: 

cc/ @yansern @sirreal
